frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,108 @@
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('groupIncome.contracts/invite/mailRecord/create', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('groupIncome.contracts/invite/groupRecord/create', inviteToMailbox.toHash(), memberName, groupId)
+
+          if (this.isProposal) {
+            let latest = await backend.latestHash(groupId)
+            let proposal = new Events.HashableGroupProposal({
+              // for proposal template selection in Vote.vue
+              type: 'invitationProposal', // TODO: constants for different proposal types
+              // calculate the voting threshold from the group data
+              threshold: this.currentGroupState.memberApprovalThreshold,
+              candidate: memberName,
+              actions: [
+               { contractId: mailbox, action: JSON.stringify(inviteToMailbox.toObject()) },
+               { contractId: groupId, action: JSON.stringify(inviteToGroup.toObject()) }
+              ],
+              initiator: this.loggedIn.name,
+              initiationDate: new Date().toString()
+            }, latest)
+            await backend.publishLogEntry(groupId, proposal)
+          } else {
+         // TODO: place these in SBP calls, consider using ArchivedTransactionQueue
+         // NOTE: avoiding use of ArchivedTransactionQueue is better than using it. Here for example,
+         //       we could make it so that a user can be invited to a group multiple times, and the system
+         //       just accepts/cares about the most recent invite only and forgets the previous one
+            await backend.publishLogEntry(groupId, inviteToGroup)
```

Comment:
Is it me or are these comments not aligned with code below? (I.e. a few extra spaces needed in front of the `// `?)